app/experimenter/nimbus-ui/src/components/FormAudience/index.test.tsx

```diff
@@ -31,6 +72,114 @@ describe("FormAudience", () => {
+    await waitFor(() => {
+      expect(screen.queryByTestId("FormAudience")).toBeInTheDocument();
+    });
+    const submitButton = screen.getByTestId("submit-button");
+    expect(submitButton).toHaveTextContent("Save");
+    await act(async () => {
+      fireEvent.click(submitButton);
+    });
+    expect(onSubmit).toHaveBeenCalled();
+    const [[submitData]] = onSubmit.mock.calls;
+    expect(submitData).toEqual({
+      channels: MOCK_EXPERIMENT.channels,
+      firefoxMinVersion: MOCK_EXPERIMENT.firefoxMinVersion,
+      targetingConfigSlug: MOCK_EXPERIMENT.targetingConfigSlug,
+      populationPercent: "" + MOCK_EXPERIMENT.populationPercent,
+      totalEnrolledClients: "" + MOCK_EXPERIMENT.totalEnrolledClients,
+      proposedEnrollment: "" + MOCK_EXPERIMENT.proposedEnrollment,
+      proposedDuration: "" + MOCK_EXPERIMENT.proposedDuration,
+    });
+  });
+
+  it("does not call onSubmit when submitted while loading", async () => {
+    const onSubmit = jest.fn();
+    render(<Subject {...{ onSubmit, isLoading: true }} />);
+    await waitFor(() => {
+      expect(screen.queryByTestId("FormAudience")).toBeInTheDocument();
+    });
+    const form = screen.getByTestId("FormAudience");
+    await act(async () => {
+      fireEvent.submit(form);
+    });
+    expect(onSubmit).not.toHaveBeenCalled();
+  });
+
+  it("marks input fields as invalid when blank", async () => {
+    const onSubmit = jest.fn();
+    const { container } = render(<Subject {...{ onSubmit }} />);
+    await waitFor(() => {
+      expect(screen.queryByTestId("FormAudience")).toBeInTheDocument();
+    });
+
+    for (const fieldName of [
+      "populationPercent",
+      "totalEnrolledClients",
+      "proposedEnrollment",
+      "proposedDuration",
+      "firefoxMinVersion",
+      "targetingConfigSlug",
+    ]) {
+      await act(async () => {
+        const field = screen.getByTestId(fieldName);
+        fireEvent.click(field);
+        fireEvent.change(field, { target: { value: "" } });
+        fireEvent.blur(field);
+      });
+
+      expect(
+        container.querySelector(`.invalid-feedback[data-for=${fieldName}`),
+      ).toHaveTextContent("This field may not be blank.");
+    }
+  });
+
+  it("marks multi-select channels as invalid when none are selected", async () => {
+    const onSubmit = jest.fn();
+    const { container } = render(<Subject {...{ onSubmit }} />);
+    await waitFor(() => {
+      expect(screen.queryByTestId("FormAudience")).toBeInTheDocument();
+    });
+    const channelsSelect = screen.getByLabelText("Channels");
+    await selectEvent.clearAll(channelsSelect);
+    const submitButton = screen.getByTestId("submit-button");
+    expect(submitButton).toHaveTextContent("Save");
+    await act(async () => {
+      fireEvent.click(submitButton);
+    });
+    expect(
+      container.querySelector(".invalid-feedback[data-for=channels"),
+    ).toHaveTextContent("At least one channel must be selected");
   });
```

Comment:
Figuring out how to test this multi-select field took me much longer than I expected and now looks underwhelmingly simple 😅